rmw_fastrtps_shared_cpp/src/participant.cpp

```diff
@@ -161,19 +163,78 @@ rmw_fastrtps_shared_cpp::create_participant(
   eprosima::fastdds::dds::DomainParticipantQos domainParticipantQos =
     eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->get_default_participant_qos();
 
+  // Configure discovery
+  switch (discovery_options->automatic_discovery_range) {
```

Comment:
Additional note: I think we should decide whether to override the value to `RMW_AUTOMATIC_DISCOVERY_RANGE_SYSTEM_DEFAULT` when `RMW_FASTRTPS_USE_QOS_FROM_XML=1`.

Whatever we decide, should be documented on the table [here](https://github.com/ros2/rmw_fastrtps#full-qos-configuration)

Reply:
Assuming it's possible for an XML to not contain any participant discovery related settings, I would recommend `RMW_FASTRTPS_USE_QOS_FROM_XML` not change the automatic discovery range so that a user can combine both options.

The intent here was the user would set `RMW_AUTOMATIC_DISCOVERY_RANGE_SYSTEM_DEFAULT` to prevent this code from overriding any participant discovery settings that might be configured out-of-band, such as via an xml file.

Added documentation in a5dae76